jmbitcoin/jmbitcoin/secp256k1_ecies.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/python
+
+import coincurve as secp256k1
+import base64
+import hmac
+import hashlib
+import pyaes
+import os
+import jmbitcoin as btc
+
+ECIES_MAGIC_BYTES = b'BIE1'
+
+class ECIESDecryptionError(Exception):
+    pass
+
+# AES primitives. See BIP-SNICKER for specification.
+def aes_encrypt(key, data, iv):
+    encrypter = pyaes.Encrypter(
+        pyaes.AESModeOfOperationCBC(key, iv=iv))
+    enc_data = encrypter.feed(data)
+    enc_data += encrypter.feed()
+
+    return enc_data
+
+def aes_decrypt(key, data, iv):
+    decrypter = pyaes.Decrypter(
+        pyaes.AESModeOfOperationCBC(key, iv=iv))
+    try:
+        dec_data = decrypter.feed(data)
+        dec_data += decrypter.feed()
+    except ValueError:
+        # note decryption errors can come from PKCS7 padding errors
+        raise ECIESDecryptionError()
+    return dec_data
+
+def ecies_encrypt(message, pubkey):
+    """ Take a privkey in raw byte serialization,
+    and a pubkey serialized in compressed, binary format (33 bytes),
+    and output the shared secret as a 32 byte hash digest output.
+    The exact calculation is:
+    shared_secret = SHA256(privkey * pubkey)
+    .. where * is elliptic curve scalar multiplication.
+    See https://github.com/bitcoin/bitcoin/blob/master/src/secp256k1/src/modules/ecdh/main_impl.h
```

Comment:
this points to `secp256k1_ecdh`, FYI, python-bitcointx has `privkey.ECDH(pub)` method implemented here https://github.com/Simplexum/python-bitcointx/blob/122c74f01d745caeeb1548c02f6458ec667ea358/bitcointx/core/key.py#L232-L251 based on `secp256k1_ecdh` from libsecp256k1. It requires that the functionality is actually enabled in the compiled library, but since you're supplying your own dynamic lib, this shouldn't be a problem

Reply:
Oh, scratch that, just saw the 'this is *not* ECDH as in the secp256k1_ecdh module' comment